cudapoa/benchmarks/common/utils.hpp

```diff
@@ -93,6 +95,139 @@ inline std::string parse_golden_value_file(const std::string& filename)
     std::getline(infile, line);
     return line;
 }
+
+/// \brief Create a small set of batch-sizes to increase GPU parallelization.
+///        Input windows are grouped based on their sizes. Similarly sized windows are grouped in the same bin.
+///        Separating smaller windows from large ones allows to launch a larger number of windows concurrently.
+///        This increase in parallelization translates to faster runtime
+///        This multi-batch strategy is particularly useful when input windows sizes display a large variance.
+///
+/// \param list_of_batch_sizes [out] a set of batch-sizes, covering all input windows
+/// \param list_of_windows_per_batch [out] corresponding windows per batch-size
+/// \param windows [in] vector of input windows
+/// \param banded_alignment [in] flag indicating whether banded-alignment or full-alignment is going to be used, default is banded-alignment
+/// \param msa_flag [in] flag indicating whether MSA or consensus is going to be computed, default is consensus
+/// \param bins_capacity [in] pointer to vector of bins used to create separate different-sized windows, if null as input, a set of default bins will be used
+/// \param gpu_memory_usage_quota [in] portion of GPU available memory that will be used for compute each cudaPOA batch, default 0.9
+/// \param mismatch_score [in] mismatch score, default -6
+/// \param gap_score [in] gap score, default -8
+/// \param match_score [in] match core, default 8
+void get_multi_batch_sizes(std::vector<BatchSize>& list_of_batch_sizes,
```

Comment:
let's move this function into the cudapoa library API. that way it can be accessible through the public API.

location - `cudapoa/include/claragenomics/cudapoa/utils.hpp` for the function signature

couple of updates needed to the API - 
1. the `windows` argument should be updated to `poa_groups` and should accept `const std::vector<Group>&` with Group as defined in https://github.com/clara-parabricks/ClaraGenomicsAnalysis/blob/dev-v0.5.0/cudapoa/include/claragenomics/cudapoa/batch.hpp#L50
2. `list_of_windows_per_batch` should be `list_of_groups_per_batch`, and the documentation should reflect that accordingly

the implementation can be in `cudapoa/src/utils.cpp`